R/generateIIASASubmission.R

```diff
@@ -97,82 +147,62 @@ generateIIASASubmission <- function(mifs = ".", mapping = NULL, model = NULL, #
            " before 1.111.0 on 2023-05-26, please use piamInterfaces version 0.9.0 or earlier, see PR #128.")
   }
 
-  # generate mapping file, if it doesn't exist yet
-  if (length(mapping) > 0 || is.null(mappingFile)) {
-    mapData <- generateMappingfile(templates = mapping, outputDirectory = NULL,
-                                   fileName = NULL, model = model, logFile = logFile,
-                                   iiasatemplate = iiasatemplate)[["mappings"]]
-    if (! is.null(mappingFile)) {
-      warning("mappingFile='", mappingFile, "' is ignored because you specified mapping.")
-      mappingFile <- NULL
-    }
-  } else if (length(mappingFile) == 1 && file.exists(mappingFile)) {
-    mapData <- read.csv2(mappingFile)
-  } else {
-    stop("Specify either one single 'mappingFile' or use (multiple) templates using the 'mapping' parameter.")
-  }
-
-  mapData <- mapData %>%
-    mutate(
-      !!sym("factor") := as.numeric(!!sym("factor")),
-      # this is not optimal and error-prone: we must dissect variable into variable and unit again
-      # could be avoided, if we expect mappings with variable and unit fields
-      # instead of having the unit as part of the variable name
-      !!sym("piam_unit") := unitsplit(!!sym("piam_variable"))$unit, # nolint
-      !!sym("piam_variable") := removePlus(unitsplit(!!sym("piam_variable"))$variable), # nolint
-      !!sym("Unit") := unitsplit(!!sym("Variable"))$unit, # nolint
-      !!sym("Variable") := unitsplit(!!sym("Variable"))$variable # nolint
-    )
-
-  message("\n### Generating submission file using mapping ", paste(c(mapping, mappingFile), collapse = ", "), ".")
-
   mifdata <- .setModelAndScenario(mifdata, model, removeFromScen, addToScen)
 
-  message("# Apply mapping ", mappingFile)
+  # apply mapping to data ----
+
+  message("# Apply generated mapping to data")
 
   submission <- mifdata %>%
-    filter(!!sym("period") %in% timesteps) %>%
+    filter(.data$period %in% timesteps) %>%
     mutate(
-      !!sym("variable") := removePlus(str_trim(!!sym("variable"))),
-      !!sym("unit") := str_trim(!!sym("unit"))
+      "piam_variable" = removePlus(str_trim(.data$variable)),
+      "piam_unit" = str_trim(.data$unit)
       ) %>%
+    select(-c("variable", "unit")) %>%
     distinct() %>%
-    inner_join(mapData, by = c("variable" = "piam_variable", "unit" = "piam_unit"),
+    inner_join(mapData, by = c("piam_variable", "piam_unit"),
                relationship = "many-to-many") %>%
-    mutate(
-      !!sym("value") := ifelse(is.na(!!sym("factor")), !!sym("value"), !!sym("factor") * !!sym("value"))
-    ) %>%
-    select("model", "scenario", "region", "period", "variable" = "Variable", "unit" = "Unit", "value")
-
+    mutate("value" = .data$piam_factor * .data$value) %>%
+    select("model", "scenario", "region", "period", "variable", "unit", "value")
 
   submission <- aggregate(value ~ model + region + scenario + period + variable + unit, data = submission, FUN = "sum")
```

Comment:
I noticed the data is completely unsorted. I suggest to add a `%>% quitteSort()` here.

Reply:
Has been added